FrEee/Scripts/AI/Vanguard/MinistryOfColonization.csx

```diff
@@ -0,0 +1,296 @@
+    /// Creates any build plans that need creating. 
+    /// </summary>
+    public void CreateBuildPlans()
+    {
+        foreach(var plan in PlanManager.CurrentPlans.ColonizationPlans.Where(x => x.AwaitingBuild && x.ShipBuildPlan == null))
+        {
+            var buildPlan = new ShipBuildPlan()
+            {
+                IsComplete = false,
+                RequestPlanId = plan.PlanId,
+                SentOrder = false,
+                Role = plan.Planet.ColonizationAbilityName == "Colonize Planet - Gas Giant" ? "Colonize Planet - Gas" : plan.Planet.ColonizationAbilityName, 
+                Priority = plan.Priority, 
+                ConstructionQueue = plan.SourcePlanet?.ConstructionQueue
+            };
+            buildPlan.AssignId();
+            plan.ShipBuildPlanId = buildPlan.PlanId;
+            buildPlan.RequestPlanId = plan.PlanId; 
+            PlanManager.CurrentPlans.ShipBuildPlans.Add(buildPlan); 
+        }
+    }
+
+    /// <summary>
+    /// Updates any plan that has not yet truly started. 
+    /// </summary>
+    public void UpdateUnstartedPlans()
+    {
+        if (updatedPlans)
+            return; 
+        foreach(var plan in PlanManager.CurrentPlans.ColonizationPlans.Where(x => x.AwaitingBuild && x.ShipBuildPlanId == null))
+        {
+            //someone else settled this world, so cancel the plan. 
+            if (plan.Planet.Colony != null)
+            {
+                plan.IsComplete = true;
+                if (plan.ShipBuildPlan != null)
+                {
+                    plan.ShipBuildPlan.IsComplete = true;
+                }
+                continue;
+            }
+
+
+            int distance = int.MaxValue;
+            Planet sourcePlant;
+            foreach (var colony in Empire.ColonizedPlanets.Where(x => x.ConstructionQueue.IsSpaceYardQueue))
+            {
+                var curdistance = Pathfinder.EstimateDistance(colony.Sector, plan.Planet.Sector, Empire);
+                if (curdistance < distance)
+                {
+                    plan.SourcePlanetId = colony.ID;
+                    distance = curdistance;
+                    sourcePlant = colony;
+                }
+            }
+            //Lost connection to this world? set it to min priority. 
+            if (distance == int.MaxValue)
+            {
+                plan.Priority = int.MinValue; 
+                continue; 
+            }
+
+            CalculateScore(plan); 
+        }
+        updatedPlans = true; 
+    }
+
+    /// <summary>
+    /// Calculates the priority of the plan. 
+    /// </summary>
+    /// <param name="colonyPlan"></param>
+    void CalculateScore(ColonizationPlan colonyPlan)
+    {       
+        var planet = colonyPlan.Planet;
+        colonyPlan.Priority = 100; //initial value; 
+
+
+        colonyPlan.Priority -= Pathfinder.EstimateDistance(colonyPlan.SourcePlanet.Sector, planet.Sector, Empire);
+        if (planet.StarSystem.HasAbility(Empire, "Spaceport"))
```

Comment:
Might want to check to see if the population on the colony ship has the Natural Merchants trait and add 100 in that case (instead of if the system has a spaceport) as well.

Reply:
Colony ship doesn't even exist at this point, let alone have population on it. This is just working out how important the planet is to colonise, so natural merchants don't really factor into it. Eventually, I should probably add some score based on resource values.

Reply:
OK, I guess this is good enough for a first pass then!